examples/custom_shader/custom_pass.rs

```diff
@@ -0,0 +1,358 @@
+use derivative::Derivative;
+use glsl_layout::*;
+use std::path::PathBuf;
+
+lazy_static::lazy_static! {
+    static ref VERTEX_SRC: SpirvShader = PathBufShaderInfo::new(
+        PathBuf::from(concat!(env!("CARGO_MANIFEST_DIR"), "/examples/assets/shaders/vertex/custom.vert")),
+        ShaderKind::Vertex,
+        SourceLanguage::GLSL,
+        "main",
+    ).precompile().unwrap();
+
+    static ref VERTEX: SpirvShader = SpirvShader::new(
+        (*VERTEX_SRC).spirv().unwrap().to_vec(),
+        (*VERTEX_SRC).stage(),
+        "main",
+    );
+
+    static ref FRAGMENT_SRC: SpirvShader = PathBufShaderInfo::new(
+        PathBuf::from(concat!(env!("CARGO_MANIFEST_DIR"), "/examples/assets/shaders/fragment/custom.frag")),
+        ShaderKind::Fragment,
+        SourceLanguage::GLSL,
+        "main",
+    ).precompile().unwrap();
+
+    static ref FRAGMENT: SpirvShader = SpirvShader::new(
+        (*FRAGMENT_SRC).spirv().unwrap().to_vec(),
+        (*FRAGMENT_SRC).stage(),
+        "main",
+    );
+}
+
+/// Draw triangles.
+#[derive(Clone, Debug, PartialEq, Derivative)]
+#[derivative(Default(bound = ""))]
+pub struct DrawCustomDesc;
+
+impl DrawCustomDesc {
+    /// Create instance of `DrawCustomDesc` render group
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+impl<B: Backend> RenderGroupDesc<B, World> for DrawCustomDesc {
+    fn build(
+        self,
+        _ctx: &GraphContext<B>,
+        factory: &mut Factory<B>,
+        _queue: QueueId,
+        _world: &World,
+        framebuffer_width: u32,
+        framebuffer_height: u32,
+        subpass: hal::pass::Subpass<'_, B>,
+        _buffers: Vec<NodeBuffer>,
+        _images: Vec<NodeImage>,
+    ) -> Result<Box<dyn RenderGroup<B, World>>, failure::Error> {
+        let env = DynamicUniform::new(factory, pso::ShaderStageFlags::VERTEX)?;
+        let vertex = DynamicVertexBuffer::new();
+
+        let (pipeline, pipeline_layout) = build_custom_pipeline(
+            factory,
+            subpass,
+            framebuffer_width,
+            framebuffer_height,
+            vec![env.raw_layout()],
+        )?;
+
+        Ok(Box::new(DrawCustom::<B> {
+            pipeline,
+            pipeline_layout,
+            env,
+            vertex,
+            vertex_count: 0,
+        }))
+    }
+}
+
+/// Draws triangles to the screen.
+#[derive(Debug)]
+pub struct DrawCustom<B: Backend> {
+    pipeline: B::GraphicsPipeline,
+    pipeline_layout: B::PipelineLayout,
+    env: DynamicUniform<B, CustomUniformArgs>,
+    vertex: DynamicVertexBuffer<B, CustomArgs>,
+    vertex_count: usize,
+}
+
+impl<B: Backend> RenderGroup<B, World> for DrawCustom<B> {
+    fn prepare(
+        &mut self,
+        factory: &Factory<B>,
+        _queue: QueueId,
+        index: usize,
+        _subpass: hal::pass::Subpass<'_, B>,
+        world: &World,
+    ) -> PrepareResult {
+        let (triangles,) = <(ReadStorage<'_, Triangle>,)>::fetch(world);
+
+        //Get our scale value
```

Comment:
Space missing after `//`. Also (some of) these comments doesn't say much thing, so they're better removed.